routers/api-authorization.js

```diff
@@ -0,0 +1,45 @@
+
+const {parse} = require("graphql");
+
+const PUBLIC = 'public';
+
+// escape public query 
+const getAPINameFromReq = (req) => {
+    if(req.body && req.body.query){
+        const parsedQuery = parse(req.body.query);
+        const api_name= parsedQuery.definitions.find(
+            (definition) => definition.kind === 'OperationDefinition'
+        ).selectionSet?.selections[0]?.name?.value;
+        return api_name; 
+    }
+    else return null
+}
+
+function extractAPINames(schema, api_type = PUBLIC){
+    const fields = schema._queryType?._fields;
+    const public_api_list = Object.values(fields).filter(
+        (f) => (f.astNode.directives.length > 0) && (f.astNode.directives[0].name?.value === api_type)
+      );
+    return public_api_list.map( (api) => api.name )
+}
+
+async function apiAuthorization(req, authenticationService, userInitializationService, public_api_list) {
+    try {
+        if (getAPINameFromReq(req) in public_api_list) return;
```

Comment:
The array check here needs to be updated. We can use "public_api_list.includes(getAPINameFromReq(req))" The "in" keyword is used to check if an object has a property in javascript 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in

Reply:
Used includes.